frontend/javascripts/oxalis/view/right-menu/advanced_search_popover.js

```diff
@@ -0,0 +1,150 @@
+// @flow
+import { Icon, Input, Tooltip, Popover } from "antd";
+import * as React from "react";
+import _ from "lodash";
+import ButtonComponent from "oxalis/view/components/button_component";
+
+import Shortcut from "libs/shortcut_component";
+
+const InputGroup = Input.Group;
+
+type Props<S> = {
+  data: { +[number]: S } | Array<S>,
+  searchKey: $Keys<S>,
+  onSelect: S => void,
+  children: *,
+  provideShortcut?: boolean,
+};
+
+type State = {
+  isVisible: boolean,
+  searchQuery: string,
+};
+
+export default class AdvancedSearchPopover<S: Object> extends React.PureComponent<Props<S>, State> {
+  constructor() {
+    super();
+    this.state = {
+      isVisible: false,
+      searchQuery: "",
+    };
+    this.availableOptions = [];
+  }
+
+  availableOptions: { +[number]: S } | Array<S> = [];
+  currentPosition: number = 0;
+
+  selectNextOptionWithOffset = (offset: number) => {
+    const numberOfAvailableOptions = _.values(this.availableOptions).length;
+    if (numberOfAvailableOptions === 0) {
+      return;
+    }
+    this.currentPosition = (this.currentPosition + offset) % numberOfAvailableOptions;
+    if (this.currentPosition < 0) {
+      this.currentPosition = numberOfAvailableOptions + this.currentPosition;
+    }
+    this.props.onSelect(_.values(this.availableOptions)[this.currentPosition]);
+  };
+
+  selectNextOption = () => {
+    this.selectNextOptionWithOffset(1);
+  };
+
+  selectPreviousOption = () => {
+    this.selectNextOptionWithOffset(-1);
+  };
+
+  onSearchQueryChanged = (searchQuery: string) => {
+    this.currentPosition = -1;
+    this.setState({ searchQuery });
+    if (searchQuery === "") {
+      this.availableOptions = [];
+    } else {
+      this.availableOptions = _.values(this.props.data).filter(
```

Comment:
I think this is not the correct spot to calculate the availableOptions.
There is a bug in this search popover, that's caused by this. When searching for term x, then closing the popover, removing/adding some trees and then opening the popover again, the availableOptions are still the old ones. They are only updated when the search term is changed.
Ideally the availableOptions should be calculated on demand in the render method I'd say (memoized with props.data, searchQuery) being the key), but unfortunately the memoization wouldn't work very well I think, because changing the activeNodeId/GroupId will change props.data (because `getTreeAndTreeGroupList` in the `trees_tab_view` is called every time the skeletonTracing changes). I have an idea how to fix this, but let's talk about that in a quick call :)